### PR TITLE
PSI: Add TomlPsiFactory

### PR DESCRIPTION
**Pull request by mibac138**
1st out of 4 groundwork PRs thanks to which there will be in turn added support for newer crate version intention and an error annotator detecting common mistakes in dependencies. The PRs are divided so that it's 
easier to review them.

Final effect once the groundwork and 2 feature (new version intention, error checking) PRs will be merged:
version-404
name-404

update

### CODE REVIEW

File: intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        createFile(code).descendantOfTypeStrict()
+
+    // Copied from org.rust.lang.core.psi.ext as it's not available here
+    private inline fun <reified T : PsiElement> PsiElement.descendantOfTypeStrict(): T? =
+        PsiTreeUtil.findChildOfType(this, T::class.java, /* strict */ true)
+
+    fun createValue(value: String): TomlValue =
+        // If you're creating a string value, like `serde = "1.0.90"` make sure that the `value` parameter actually
+        // contains the quote in the beginning and the end. E.g.: `createValue("\"1.0.90\"")`
+        createFromText<TomlKeyValue>("dummy = $value")?.value ?: error("Failed to create TomlValue")
```
Comment (mchernyavsky) [inline intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt]:
```suggestion
        createFromText("dummy = $value") ?: error("Failed to create TomlValue")
```

File: intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    fun createKeyValue(key: String, value: String): TomlKeyValue =
+        createFromText("$key = $value") ?: error("Failed to create TomlKeyValue")
+
+    fun createTable(name: String): TomlTable =
+        createFromText("[$name]") ?: error("Failed to create TomlTableHeader")
+
+    fun createTableHeader(name: String): TomlTableHeader =
+        createFromText<TomlTable>("[$name]")?.header ?: error("Failed to create TomlTableHeader")
+
+    fun crateArray(contents: String): TomlArray =
```
Comment (mchernyavsky) [inline intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt]:
```suggestion
    fun createArray(contents: String): TomlArray =
```

File: intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    private inline fun <reified T : PsiElement> PsiElement.descendantOfTypeStrict(): T? =
+        PsiTreeUtil.findChildOfType(this, T::class.java, /* strict */ true)
+
+    fun createValue(value: String): TomlValue =
+        // If you're creating a string value, like `serde = "1.0.90"` make sure that the `value` parameter actually
+        // contains the quote in the beginning and the end. E.g.: `createValue("\"1.0.90\"")`
+        createFromText<TomlKeyValue>("dummy = $value")?.value ?: error("Failed to create TomlValue")
+
+    fun createKey(key: String): TomlKey =
+        createFromText<TomlKeyValue>("$key = \"dummy\"")?.key ?: error("Failed to create TomlKey")
```
Comment (mchernyavsky) [inline intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt]:
```suggestion
        createFromText("$key = \"dummy\"") ?: error("Failed to create TomlKey")
```

File: intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        createFromText(text) ?: error("Failed to create TomlKeyValue")
+
+    fun createKeyValue(key: String, value: String): TomlKeyValue =
+        createFromText("$key = $value") ?: error("Failed to create TomlKeyValue")
+
+    fun createTable(name: String): TomlTable =
+        createFromText("[$name]") ?: error("Failed to create TomlTableHeader")
+
+    fun createTableHeader(name: String): TomlTableHeader =
+        createFromText<TomlTable>("[$name]")?.header ?: error("Failed to create TomlTableHeader")
```
Comment (mchernyavsky) [inline intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt]:
```suggestion
        createFromText("[$name]") ?: error("Failed to create TomlTableHeader")
```

File: intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    fun crateArray(contents: String): TomlArray =
+        createFromText<TomlKeyValue>("dummy = [$contents]")?.value as TomlArray? ?: error("Failed to create TomlArray")
```
Comment (mchernyavsky) [inline intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt]:
```suggestion
        createFromText("dummy = [$contents]") ?: error("Failed to create TomlArray")
```

File: intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    fun createInlineTable(contents: String): TomlInlineTable =
+        createFromText<TomlKeyValue>("dummy = {$contents}")?.value as TomlInlineTable? ?: error("Failed to create TomlInlineTable")
```
Comment (mchernyavsky) [inline intellij-toml/src/main/kotlin/org/toml/lang/psi/TomlPsiFactory.kt]:
```suggestion
        createFromText("dummy = {$contents}") ?: error("Failed to create TomlInlineTable")
```

**Comment by mibac138** [comment]
Thank you for the tests :). I've applied the changes you suggested

**Comment by mchernyavsky** [comment]
Thanks!
bors r+

**Pull request merged**